libgearman-server/plugins/queue/redis/queue.cc

```diff
@@ -202,7 +202,7 @@ static gearmand_error_t _hiredis_add(gearman_server_st *, void *context,
   build_key(key, unique, unique_size, function_name, function_name_size);
   gearmand_log_debug(GEARMAN_DEFAULT_LOG_PARAM, "hires key: %u", (uint32_t)key.size());
 
-  redisReply *reply= (redisReply*)redisCommand(queue->redis(), "SET %b %b", &key[0], key.size(), data, data_size);
+  redisReply *reply= (redisReply*)redisCommand(queue->redis(), "SET %s %b", &key[0], data, data_size);
```

Comment:
I'm a wee bit concerned that we're trading a nice safe fixed length string for a hopefully null terminated string. Is there some reason for that?